### PR TITLE
Fix issue #906

### PR DESCRIPTION
**Pull request by darxis**
Fixes issue #906. Please review.

### CODE REVIEW

**Review by vb2ae — CHANGES_REQUESTED** [review]
Charges look good.  Will need to merge in pr #891 once I get a review and merged into master.  It will fix the build issues.

File: src/Caliburn.Micro.Core/ConductorBaseWithActiveItem.cs [hunk 1]
```diff
@@ -39,12 +39,12 @@ object IHaveActiveItem.ActiveItem
...
         protected virtual async Task ChangeActiveItemAsync(T newItem, bool closePrevious, CancellationToken cancellationToken)
         {
-            await ScreenExtensions.TryDeactivateAsync(_activeItem, closePrevious, cancellationToken);
-            newItem = EnsureItem(newItem);
-
             _activeItem = newItem;
             NotifyOfPropertyChange(nameof(ActiveItem));
 
+            await ScreenExtensions.TryDeactivateAsync(_activeItem, closePrevious, cancellationToken);
+            newItem = EnsureItem(newItem);
```
Comment (vb2ae) [inline src/Caliburn.Micro.Core/ConductorBaseWithActiveItem.cs:46]: think you need to move this above _activeItem = newItem;

Comment (darxis) [reply src/Caliburn.Micro.Core/ConductorBaseWithActiveItem.cs]: You're right, I overlooked this call. Commited [10180d1](https://github.com/Caliburn-Micro/Caliburn.Micro/pull/907/commits/10180d1d3b1d8a71716ecd919debe60b4eb957ea)

Comment (vb2ae) [reply src/Caliburn.Micro.Core/ConductorBaseWithActiveItem.cs]: Are you still having issues with INotifyPropertyChanged?

Comment (darxis) [reply src/Caliburn.Micro.Core/ConductorBaseWithActiveItem.cs]: Do you mean the `NotifyOfPropertyChange(nameof(ActiveItem));` call?

I meant that this call should have prevented the bug on the unfixed master branch because it should trigger a UI refresh, but it have not and I don't know why. Please check my [bug-async-calls-in-ondeactivateasync](https://github.com/darxis/Caliburn.Micro/tree/bug-async-calls-in-ondeactivateasync) branch where I modified the `Caliburn.Micro.Tutorial.Wpf` example to reproduce this bug.

Comment (darxis) [reply src/Caliburn.Micro.Core/ConductorBaseWithActiveItem.cs]: > I think it what happens is the following:
> 1. When you click the 'Some other view' tab the Dispatcher (or some other thread) sets the `ActiveItem` with the `SomeOtherViewModel` instance. 
> 2. The `ActiveItem` setter call returns before the `OnDeactivateAsync` method is fully executed as the `ActivateItemAsync` call is not awaited so the `_activeItem` is not changed because it is updated after `OnDeactivateAsync` is completed. That means that the `ActiveItem` setter call returns but the `ActiveItem` would return the old viewmodel until the `OnDeactivateAsync` completes and `_activeItem` is changed.
> 3. The thread/code that set the `ActiveItem` marks that property as "Dirty" and triggers refreshing of that property before the underneath `_activeItem` field is changed - you can check that by putting a breakpoint on the `ActiveItem` setter.

But then the `OnDeactivateItem` finally completes asynchronously and the `_activeItem` is set and the `NotifyOfPropertyChange(nameof(ActiveItem));` should work, but it is not working and the UI still displays the old view.

Comment (vb2ae) [reply src/Caliburn.Micro.Core/ConductorBaseWithActiveItem.cs]: Ok I will look

Comment (darxis) [reply src/Caliburn.Micro.Core/ConductorBaseWithActiveItem.cs]: It seems like whatever the thing that should handle the `PropertyChanged(ActiveItem)` event and refresh the UI is just ignoring this event.

Comment (darxis) [reply src/Caliburn.Micro.Core/ConductorBaseWithActiveItem.cs]: > Okay, I managed to find out why the `TabControl` doesn't update the UI despite `PropertyChanged(ActiveItem)` being called. It is because when you click the second tab on the UI to show it then it is the `TabControl` that sets the `ActiveItem` and internally `TabControl` will remember this newly selected tab as the active tab. And later when the `PropertyChanged(ActiveItem)` is called the `TabControl` compares if the active tab it remembered is different than the new one because if that's the case there is no need to do anything. In this case it is the same tab that's why the UI doesn't update to the newly selected tab view.
> 
> So `TabControl` has the correct tab remembered but it displays the view for the old tab as `ActiveItem/_activeItem` is not updated yet because it is updated asynchronously later.
> 
> Also this bug doesn't occur if you activate the other tab programatically by calling `ActiveItem = _someOtherViewModel` from within the `Conductor`, because TabControl doesn't know the new tab yet, it get to know it only when the `NotifyOfPropertyChange(nameof(ActiveItem))` is called and then it refreshes the UI with the correct view.
> 
> TL;DR - everything is fine with the `NotifyOfPropertyChange(nameof(ActiveItem));` call and `ActiveItem` should set its backed field `_activeItem` synchronously before exiting the setter, not asynchronously within a new `Task` - the PR I submitted fixes this behavior.

**Comment by vb2ae** [comment]
pr #891 is merged in can you merge or rebase this to the latest in the master branch?  I will approve and merge in

**Review by vb2ae — APPROVED** [review]

**Comment by vb2ae** [comment]
@darxis thanks It is merged in. I need to make a few changes to the build github action. Did not think about setting up the github feed credentials in the build from a fork would fail because the secrets are not available

**Comment by darxis** [comment]
@vb2ae Thanks. Can we make a new v4 release with this bugfix? I am still using v4 and it would be nice to have it fixed in as v4 (v4.0.213?) as v5 is still in beta.

**Comment by vb2ae** [comment]
There is 4.0.212 branch we can merge your fix into.   Not sure it will build because it uses some unsuported platforms
 https://github.com/Caliburn-Micro/Caliburn.Micro/tree/V4_0_212
